### DISCUSSION TITLE
Cross-compilation Linux->Windows

### DISCUSSION THREAD

**Discussion by DrDub** [zephir-lang/zephir#2359] [Ideas]
Hello!

I'm interested in generating an extension .DLL from a Linux box. (Why? Because it is good to be able to support Windows users without having to install windows altogether.)

Using MingW64 and running

```bash
./configure --prefix=/usr/bin/i686-w64-mingw32 --host=i686-w64-mingw32 --build=i686-linux
```

On a Debian box works and produces a Makefile that attempts to compile code. But of course the extension code is targeting Linux so it doesn't work.

I would like to hear your ideas of how Zephir can be run under Linux to generate the Windows stub.

( Assuming the machine has the Windows php dev package from https://windows.php.net/download/ )

Any help will be most welcomed!

**Comment by Jeckerson** [comment]
Try to configure win environment how it was done here - https://github.com/zephir-lang/zephir/blob/cc7ecd35745594710913edab1afe5c6d82f52932/.github/workflows/build-win-ext/action.yml#L45

**Comment by DrDub** [comment]
Thanks for your kind reply!

It turns out that PHP is built using MSVC and it doesn't support MingGW as a compiler. That means that extensions need to adhere to `__vectorcall` calling conventions for all the API marked as "`ZEND_FASTAPI`" (which are tons of them). But `__vectorcall` is not yet supported in GCC:

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=89485

(This appears in the compiler errors as "undefined reference to `__imp_new_array`" because the `php8ts.lib` contains the symbol `_new_array@@8` instead of `_new_array`, the mangled `@@8` refers to the `__vectorcall`.)
